### PR TITLE
chore(papyrus_network): create readme and run scripts for stress test

### PR DESCRIPTION
**Pull request by eitanm-starkware**

### CODE REVIEW

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: all files reviewed, 17 unresolved discussions (waiting on @eitanm-starkware)

___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 1 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_OWWCZ5W_L7UOKxM:-OD0_OWWCZ5W_L7UOKxN:bjjmaxg) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L1)):*
> ```Shell
> #!/bin/bash
> ```

Add TODO to support a generic amount of instances and not 5
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 3 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0aZ9G2DCheRHzP4Rq:-OD0aZ9HDnkG7IFeHfg6:bd74cgz) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L3)):*
> ```Shell
> #!/bin/bash
> 
> if [[ -z "$BASE_INSTANCE_NAME" || -z "$PROJECT_ID" || -z "$BASE_PATH" || -z "$ZONE" ]]; then
> ```

No need for BASE_PATH. Just use hardcoded "~"
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 4 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0Zy-W8zJBcubBiN84:-OD0Zy-W8zJBcubBiN85:biedyb8) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L4)):*
> ```Shell
> 
> if [[ -z "$BASE_INSTANCE_NAME" || -z "$PROJECT_ID" || -z "$BASE_PATH" || -z "$ZONE" ]]; then
>     echo "Error: BASE_INSTANCE_NAME, PROJECT_ID, and PATH must be set."
> ```

PATH -> BASE_PATH
What about ZONE?
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 5 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0Zt7A9JIcOvIW-Rch:-OD0Zt7A9JIcOvIW-Rci:b-ovx15d) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L5)):*
> ```Shell
> if [[ -z "$BASE_INSTANCE_NAME" || -z "$PROJECT_ID" || -z "$BASE_PATH" || -z "$ZONE" ]]; then
>     echo "Error: BASE_INSTANCE_NAME, PROJECT_ID, and PATH must be set."
>     echo "Instance name must be set in the format 'base-instance-name-0'... 'base-instance-name-4'."
> ```

The format is not clear. Try to put it without dots
Also, use BASE_INSTANCE_NAME and not base-instance-name
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 13 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0__w95Qme3z6uFMrr:-OD0__w95Qme3z6uFMrs:b62nn0k) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L13)):*
> ```Shell
> # Loop over instances from 0 to 4
> for i in {0..4}; do
> (   
> ```

Remove spaces
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 14 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_ZM8-ssFa1DzMwjp:-OD0_ZM8-ssFa1DzMwjq:b-4fwu1l) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L14)):*
> ```Shell
> for i in {0..4}; do
> (   
>  INSTANCE_NAME="${BASE_INSTANCE_NAME}-${i}"
> ```

Indent this line
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 16 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_VjEEmJclwM5yA8A:-OD0_VjEEmJclwM5yA8B:b-rysa7n) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L16)):*
> ```Shell
>  INSTANCE_NAME="${BASE_INSTANCE_NAME}-${i}"
>     echo "Connecting to $INSTANCE_NAME..."
>     
> ```

Change this into a tab
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 1 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_vAa3k_pU4KwetC1:-OD0_vAa3k_pU4KwetC2:b-i28ncv) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L1)):*
> ```Markdown
> ## Network Stress Test
> ```

This should be one shebang
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 3 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_xZv1y9qYldOziYC:-OD0_xZv1y9qYldOziYD:b35g78s) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L3)):*
> ```Markdown
> ## Network Stress Test
> 
> # Setup and Run Stress Test
> ```

This should be two shebangs
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 15 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_miGDTvl4rEvRGhg:-OD0_miGDTvl4rEvRGhh:b-nxn7pf) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L15)):*
> ```Markdown
> 2. **Set Bootstrap Node**
> 
>     Find the internal IP of your bootstrap node in the VM instances chart on google cloud console. Paste it into the test_config.json file into the bootstrap_peer_multaddr value instead of it's placeholder.
> ```

it's -> its
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 23 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_qXoEkpXtTV5jlc0:-OD0_qXoEkpXtTV5jlc1:bk64xxs) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L23)):*
> ```Markdown
>     ```
>     gcloud compute ssh <instance-name>-0 --project <project-name> -- 'cd <path-to-repo> && sudo apt install -y git unzip clang && curl https://sh.rustup.rs -sSf | sh -s -- -y && source "$HOME/.cargo/env" && git clone https://github.com/starkware-libs/sequencer.git; cd sequencer && sudo scripts/dependencies.sh cargo build --release -p papyrus_network --bin network_stress_test'
>     ```
> ```

Add newline
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 1 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0a23jB-rsqm-1u5Wj:-OD0a23jB-rsqm-1u5Wk:b74wa9s) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L1)):*
> ```Shell
> #!/bin/bash
> ```

rename to run_broadcast_stress_test.sh
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 3 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0acEN7s_3Nz0vWKsz:-OD0acEN7s_3Nz0vWKt-:b-ifzzxv) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L3)):*
> ```Shell
> #!/bin/bash
> 
> if [[ -z "$BASE_INSTANCE_NAME" || -z "$PROJECT_ID" || -z "$BASE_PATH" || -z "$ZONE" ]]; then
> ```

Same as before (BASE_PATH -> ~)
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 5 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0aCJE8oshIGa_7Qho:-OD0aCJE8oshIGa_7Qhp:b36aixk) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L5)):*
> ```Shell
> if [[ -z "$BASE_INSTANCE_NAME" || -z "$PROJECT_ID" || -z "$BASE_PATH" || -z "$ZONE" ]]; then
>     echo "Error: BASE_INSTANCE_NAME, PROJECT_ID, ZONE, and PATH must be set."
>     echo "Instance name must be set in the format 'base-instance-name-0'... 'base-instance-name-4'."
> ```

Same as before
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 46 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0ax4m4fh1zt0RBk4Y:-OD0ax4m4fh1zt0RBk4Z:b-qdzizp) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L46)):*
> ```Shell
> 
> # Retrieve the bootstrap output from instance 0
> echo "Retrieving bootstrap output from ${BASE_INSTANCE_NAME}-0..."
> ```

Why is this handled separately and not with the rest?
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 60 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0b371C6ow-5JaCOmX:-OD0b371C6ow-5JaCOmY:bo9omv1) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L60)):*
> ```Shell
> 
> # Wait for file retrieval processes to complete
> wait
> ```

Erase this
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 62 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0b47MCCl_wGlLOYsz:-OD0b47MCCl_wGlLOYt-:bo9xroe) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L62)):*
> ```Shell
> wait
> echo "All output files retrieved."
> # #!/bin/bash
> ```

Erase everything commented out

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: all files reviewed, 18 unresolved discussions (waiting on @eitanm-starkware)

___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 118 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0exlb11yEbOigVV_7:-OD0exlb11yEbOigVV_8:bxcc9mo) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L118)):*
> ```Shell
> # # Wait for file retrieval processes to complete
> # wait
> # echo "All output files retrieved."
> ```

newline at EOF

**Review by eitanm-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: all files reviewed, 18 unresolved discussions (waiting on @ShahakShama)

___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 1 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_OWWCZ5W_L7UOKxM:-ODL8cfjD1rjyKprATtB:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L1)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Add TODO to support a generic amount of instances and not 5
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 3 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0aZ9G2DCheRHzP4Rq:-ODL8dVD4dDK6gv8W5Os:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L3)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

No need for BASE_PATH. Just use hardcoded "~"
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 4 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0Zy-W8zJBcubBiN84:-ODL8kCuBEk-RnpnWgkO:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L4)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

PATH -> BASE_PATH
What about ZONE?
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 5 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0Zt7A9JIcOvIW-Rch:-ODL9-zW7KwGQmc8MhHK:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L5)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

The format is not clear. Try to put it without dots
Also, use BASE_INSTANCE_NAME and not base-instance-name
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 13 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0__w95Qme3z6uFMrr:-ODL96VlAnJ1m83hsXI7:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L13)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Remove spaces
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 14 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_ZM8-ssFa1DzMwjp:-ODL96KAA6YmTJPA1ZRC:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L14)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Indent this line
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 16 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_VjEEmJclwM5yA8A:-ODLyGBUDaLNZomzcx8E:bi0zclp) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L16)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Change this into a tab
</blockquote></details>

Done. hopefully this will look like a tab now
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 1 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_vAa3k_pU4KwetC1:-ODLBcLa9TDLp0rAbpF_:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L1)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

This should be one shebang
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 3 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_xZv1y9qYldOziYC:-ODLBce26iDzPqp_t6yp:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L3)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

This should be two shebangs
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 15 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_miGDTvl4rEvRGhg:-ODLBeNj1GFG5pyX7WzA:b-sxyrjj) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L15)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

it's -> its
</blockquote></details>

Done. nice catch
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 23 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0_qXoEkpXtTV5jlc0:-ODLBhZwDrdQMDnExHJW:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/README.md#L23)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Add newline
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 1 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0a23jB-rsqm-1u5Wj:-ODLC-M-40CzN8teeK9J:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L1)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

rename to run_broadcast_stress_test.sh
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 3 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0acEN7s_3Nz0vWKsz:-ODLC-8F9NDRt3p2rSwB:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L3)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Same as before (BASE_PATH -> ~)
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 5 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0aCJE8oshIGa_7Qho:-ODLC4NwD14umuc_2nko:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L5)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Same as before
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 46 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0ax4m4fh1zt0RBk4Y:-ODLC5Ml0pC5K1innH92:b71ksq4) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L46)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Why is this handled separately and not with the rest?
</blockquote></details>

different cmnd
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 60 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0b371C6ow-5JaCOmX:-ODLyoxmCQbmSa3b92_J:bzfnp67) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L60)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Erase this
</blockquote></details>

this waits for all background process to finish before continuing
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 62 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0b47MCCl_wGlLOYsz:-ODLCK_L96MuYalv8ByO:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L62)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Erase everything commented out
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh` line 118 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0exlb11yEbOigVV_7:-ODLCz7e5KeWWzMQvCy8:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/run_nw_stress_test.sh#L118)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

newline at EOF
</blockquote></details>

Done.

File: crates/papyrus_network/src/bin/network_stress_test/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    For all 5 instances run:
+
+    ```
+    gcloud compute ssh <instance-name>-0 --project <project-name> -- 'cd <path-to-repo> && sudo apt install -y git unzip clang && curl https://sh.rustup.rs -sSf | sh -s -- -y && source "$HOME/.cargo/env" && git clone https://github.com/starkware-libs/sequencer.git; cd sequencer && sudo scripts/dependencies.sh cargo build --release -p papyrus_network --bin network_stress_test'
+    ```
+    
+4. **Run test**
+
+    ```
+    PROJECT_ID=<project-name> BASE_INSTANCE_NAME=<instance-name> ZONE=<zone> ./run_nw_stress_test.sh
```
[the remark this answers is not in the document]
Comment (ShahakShama) [reply crates/papyrus_network/src/bin/network_stress_test/README.md:28]: :+1:

**Review by ShahakShama — CHANGES_REQUESTED** [review]
> Reviewed 1 of 4 files at r2, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: all files reviewed, 6 unresolved discussions (waiting on @eitanm-starkware and @graphite-app[bot])

___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 5 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0Zt7A9JIcOvIW-Rch:-OFHbUwd2XNEkXXVpEHU:b-8dl8bo) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L5)):*
<details><summary><i>Previously, eitanm-starkware wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I  meant to just write everything from 0 till 4. Shouldn't be too much, right?
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 13 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0__w95Qme3z6uFMrr:-OFHah7P7lEI1d14C3i6:b-uij9z9) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L13)):*
<details><summary><i>Previously, eitanm-starkware wrote…</i></summary><blockquote>

Done.
</blockquote></details>

They're still here
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 24 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OFHamT54xCDaEN01GpG:-OFHamT54xCDaEN01GpH:b62nn0k) ([raw file](https://github.com/starkware-libs/sequencer/blob/f253496c1778f0f3e1434f74f8d94cef2b35dbae/crates/papyrus_network/src/bin/network_stress_test/README.md#L24)):*
> ```Markdown
>     gcloud compute ssh <instance-name>-0 --project <project-name> -- 'cd <path-to-repo> && sudo apt install -y git unzip clang && curl https://sh.rustup.rs -sSf | sh -s -- -y && source "$HOME/.cargo/env" && git clone https://github.com/starkware-libs/sequencer.git; cd sequencer && sudo scripts/dependencies.sh cargo build --release -p papyrus_network --bin network_stress_test'
>     ```
>     
> ```

Remove spaces
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 33 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OFHax7-CPOrDIfgTvdG:-OFHax7-CPOrDIfgTvdH:b-r7v6px) ([raw file](https://github.com/starkware-libs/sequencer/blob/f253496c1778f0f3e1434f74f8d94cef2b35dbae/crates/papyrus_network/src/bin/network_stress_test/README.md#L33)):*
> ```Markdown
> 5. **Results**
> 
>     Results are retrieved from VM instances and saved to your / directory under output.csv
> ```

That sounds very ominous. Add an output_file_path argument instead

File: crates/papyrus_network/src/bin/network_stress_test/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (eitanm-starkware) [reply crates/papyrus_network/src/bin/network_stress_test/README.md:28]: :thumbsup:

File: crates/papyrus_network/src/bin/network_stress_test/run_broadcast_stress_test.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# Wait for all commands to complete
+for pid in "${process_pids[@]}"; do
+    wait "$pid"
+done
+echo "All commands completed."
+
+# Retrieve the bootstrap output from instance 0
+echo "Retrieving bootstrap output from ${BASE_INSTANCE_NAME}-0..."
+gcloud compute ssh "${BASE_INSTANCE_NAME}-0" --project "${PROJECT_ID}" --zone "${ZONE}" --tunnel-through-iap -- "cat ${PATH_TO_REPO}/crates/papyrus_network/src/bin/network_stress_test/bootstrap_output.csv" > output.csv
+echo "Bootstrap output saved as bootstrap_output.csv."
```
[the remark this answers is not in the document]
Comment (eitanm-starkware) [reply crates/papyrus_network/src/bin/network_stress_test/run_broadcast_stress_test.sh]: :thumbsdown:

**Review by eitanm-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: all files reviewed, 6 unresolved discussions (waiting on @graphite-app[bot] and @ShahakShama)

___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 5 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0Zt7A9JIcOvIW-Rch:-OFHpd-xACkDQQRheRdZ:bl67pe1) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L5)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

I  meant to just write everything from 0 till 4. Shouldn't be too much, right?
</blockquote></details>

it's unclear. i changed it to a different message. lmk what you think
___
*[`crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh` line 13 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OD0__w95Qme3z6uFMrr:-OFHpotXCOvnvxnxcmLx:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/e4f2b5403e9543a0f728bb6412483d83260c2d9c/crates/papyrus_network/src/bin/network_stress_test/pull_stress_test.sh#L13)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

They're still here
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 24 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OFHamT54xCDaEN01GpG:-OFHqBQG1WDodx3iMo-V:b-896fix) ([raw file](https://github.com/starkware-libs/sequencer/blob/f253496c1778f0f3e1434f74f8d94cef2b35dbae/crates/papyrus_network/src/bin/network_stress_test/README.md#L24)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

Remove spaces
</blockquote></details>

Done.
___
*[`crates/papyrus_network/src/bin/network_stress_test/README.md` line 33 at r4](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-OFHax7-CPOrDIfgTvdG:-OFHqYrS8Lpp_WG8ZZEm:b-wq64b6) ([raw file](https://github.com/starkware-libs/sequencer/blob/f253496c1778f0f3e1434f74f8d94cef2b35dbae/crates/papyrus_network/src/bin/network_stress_test/README.md#L33)):*
<details><summary><i>Previously, ShahakShama wrote…</i></summary><blockquote>

That sounds very ominous. Add an output_file_path argument instead
</blockquote></details>

it is an argument via json config. i added to the message

**Review by ShahakShama — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: all files reviewed, 2 unresolved discussions (waiting on @graphite-app[bot])

**Review by ShahakShama — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347#-:-OFLVJmwBCFKRYZOHG6K:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @graphite-app[bot])

**Review by eitanm-starkware — COMMENTED** [review]
> Dismissed @graphite-app[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: all files reviewed, 1 unresolved discussion (waiting on @graphite-app[bot])

**Review by eitanm-starkware — COMMENTED** [review]
> Dismissed @graphite-app[bot] from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/2347)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @eitanm-starkware)

**Pull request merged**